docs/profiles.md

```diff
@@ -336,27 +337,80 @@ profile `my_dep` using the name `my_res2`.
 
 # Profile Attributes
 
-Attributes may be used in profiles to define secrets, such as user names and passwords, that should not otherwise be stored in plain-text in a cookbook. First specify a variable in the control for each secret, then add the secret to a YAML file located on the local machine, and then run `inspec exec` and specify the path to that Yaml file using the `--attrs` attribute.
+Attributes may be used in profiles to define secrets, such as user names and passwords, that should not otherwise be stored in plain-text in a cookbook. Attributes may be set for the whole profile in the `inspec.yml`.
 
-For example, a control:
+Attributes may contain the following options:
 
+* Use `default` to set a default value for the attribute.
+* Use `type` to restrict an attribute to a specific type (any, string, numeric, array, hash, boolean, regex).
+* Use `required` to ensure the attribute has a default value or a value from a secrets file.
+* Use `description` to set a brief description for the attribute.
+
+
+You can specify attributes in your `inspec.yml` using the `attributes` setting. For example, to add a `user` attribute for your profile:
+```YAML
+attributes:
+  - name: user
+    type: string
+    default: bob
+```
+
+Example of adding a array object of servers:
+```YAML
+attributes:
+  - name: servers
+    type: array
+    default:
+      - server1
+      - server2
+      - server3
+```
+
+To access an attribute you will use the `attribute` keyword. You can use this anywhere in your control code.
+
+For example:
 ```Ruby
-# define these attributes on the top-level of your file and re-use them across all tests!
-val_user = attribute('user', default: 'alice', description: 'An identification for the user')
-val_password = attribute('password', description: 'A value for the password')
+current_user = attribute('user')
+
+control 'system-users' do
+  describe attribute('user') do
+    it { should eq 'bob' }
+  end
+
+  describe current_user do
+    it { should eq attribute('user') }
+  end
+end
+```
 
+For sensitive data it is recomended to use a secrets YAML file located on the local machine to populate the values of attributes. A secrets file will always overwrite a attributes default value. To use the secrets file run `inspec exec` and specify the path to that Yaml file using the `--attrs` attribute.
```

Comment:
If we are going to explain this use case, I think we should also explain the parameterization use case (which is different in motivation, but identical in mechanics).